core/messaging/__init__.py

```diff
@@ -32,8 +38,35 @@ def __getitem__(self, item):
         return getattr(self, item)
 
     @property
-    def cell_indexes(self) -> str:
-        return ", ".join(self.__cell_indexes)
+    def cell_indexes(self) -> tuple[str, ...] | None:
+        return self.__cell_indexes
+
+    @classmethod
+    def __clean_cell_indexes(cls, cell_indexes) -> tuple[str, ...] | None:
+        """Remove any duplicate cell references, sorts them in column-then-row order, and ensures uppercase columns"""
+        if cell_indexes is None:
+            return None
+        if None in cell_indexes:
+            raise NotImplementedError("`cell_indexes` cannot contain None elements; must be None directly")
+        try:
+            cell_indexes = [cls.cell_index_pattern.match(cell_index).groups() for cell_index in set(cell_indexes)]
```

Comment:
Maybe rename to `cell_index_groups` or something so that we don't have the variable `cell_indexes` changing type / format throughout the function e.g., currently `cell_indexes` might enter the function as `('A1',)` and then be replaced by `[('A', '1')]` on this line, which is slightly confusing

Reply:
I've changed to `split_cell_indexes`, which I think is hopefully close enough 👍  so fixed